app/components/cases/subcase-mandatees/component.js

```diff
@@ -26,12 +25,20 @@ export default Component.extend(ManageMinisterMixin, {
 
   actions: {
     async saveChanges(mandatee, newRow) {
-      const rowToShow = await this.get('rowToShow');
+      const iseCodes = (await mandatee.get('iseCodes')).filter((item) => item);
+      const fields = (await Promise.all(iseCodes.map((iseCode) => iseCode.get('field')))).filter((item) => item);
+      const domains = await Promise.all(fields.map((field) => field.get('domain')));
+
+      const rowToShow = EmberObject.create({
+        domains: [...new Set(domains)],
+        fields: [...new Set(fields)]
+      });
+
       const mandateeRows = await this.get('mandateeRows');
       const domainsToShow = newRow.domains.map((domain) => domain.get('label')).join(', ');
       const fieldsToShow = newRow.fields.map((field) => field.get('label')).join(', ');
-      if (rowToShow) {
-        let rowToEdit = mandateeRows.find((row) => row.mandatee.id === mandatee.id);
+      let rowToEdit = mandateeRows.find((row) => row.mandatee.id === mandatee.id);
+      if (rowToShow && rowToEdit) {
```

Comment:
Another behavioural change, what was the reason?

Reply:
Wat ik denk, waarom ik die change gedaan heb in de code is omdat de add en edit modals van een minister naar 1 component gegaan zijn ipv de 2 die er al stonden.. Die gebruikten de savechanges van de mixin en niet die van de component.. Ik heb die 2 weken terug gerefactored naar 1 component en dan ook de mixin eruitgehaald waardoor die savechanges veralgemeend is naar de huidige code.

Reply:
Dus dit is een bugfix voor iets in development wat nu in deze refactor zit?

Reply:
Nee dat is een code-improvement die ik gezien heb tijdens de refactor

Reply:
Ah OK